src/test/kotlin/no/nav/familie/ef/sak/behandlingsflyt/steg/BeregnYtelseStegTest.kt

```diff
@@ -63,357 +64,423 @@ internal class BeregnYtelseStegTest {
+            assertThat(nyeAndeler[0].beløp).isEqualTo(50)
+
+            assertThat(nyeAndeler[1].stønadFom).isEqualTo(nyAndelFom)
+            assertThat(nyeAndeler[1].stønadTom).isEqualTo(nyAndelTom)
+            assertThat(nyeAndeler[1].beløp).isEqualTo(100)
+        }
+
+        @Test
+        internal fun `bytter ut siste andelen med en ny andel`() {
+            val forrigeAndelFom = LocalDate.of(2021, 1, 1)
+            val forrigeAndelTom = LocalDate.of(2021, 10, 31)
+            val forrigeAndelFom2 = LocalDate.of(2021, 11, 1)
+            val forrigeAndelTom2 = LocalDate.of(2021, 11, 30)
+            val nyAndelFom = LocalDate.of(2021, 11, 1)
+            val nyAndelTom = LocalDate.of(2021, 11, 30)
+
+            val forrigeAndeler = listOf(lagAndelTilkjentYtelse(50, forrigeAndelFom, forrigeAndelTom),
+                                        lagAndelTilkjentYtelse(70, forrigeAndelFom2, forrigeAndelTom2))
+            val beløpsperioder = listOf(lagAndelTilkjentYtelse(100, nyAndelFom, nyAndelTom))
+            val nyeAndeler = steg.slåSammenAndelerSomSkalVidereføres(beløpsperioder, lagTilkjentYtelse(forrigeAndeler), listOf())
+
+            assertThat(nyeAndeler).hasSize(2)
+            assertThat(nyeAndeler[0].stønadFom).isEqualTo(forrigeAndelFom)
+            assertThat(nyeAndeler[0].stønadTom).isEqualTo(LocalDate.of(2021, 10, 31))
+            assertThat(nyeAndeler[0].beløp).isEqualTo(50)
+
+            assertThat(nyeAndeler[1].stønadFom).isEqualTo(nyAndelFom)
+            assertThat(nyeAndeler[1].stønadTom).isEqualTo(nyAndelTom)
+            assertThat(nyeAndeler[1].beløp).isEqualTo(100)
+        }
+
+        @Test
+        internal fun `legger på opphold og ny periode`() {
+            val forrigeAndelFom = LocalDate.of(2021, 1, 1)
+            val forrigeAndelTom = LocalDate.of(2021, 10, 31)
+            val nyAndelFom = LocalDate.of(2021, 12, 1)
+            val nyAndelTom = LocalDate.of(2021, 12, 31)
+            val forrigeAndeler = listOf(lagAndelTilkjentYtelse(50, forrigeAndelFom, forrigeAndelTom))
+            val opphørsperioder = listOf(Periode(LocalDate.of(2021, 11, 1), LocalDate.of(2021, 11, 30)))
+            val beløpsperioder = listOf(lagAndelTilkjentYtelse(100, nyAndelFom, nyAndelTom))
+
+            val nyeAndeler =
+                    steg.slåSammenAndelerSomSkalVidereføres(beløpsperioder, lagTilkjentYtelse(forrigeAndeler), opphørsperioder)
+
+            assertThat(nyeAndeler).hasSize(2)
+            assertThat(nyeAndeler[0].stønadFom).isEqualTo(forrigeAndelFom)
+            assertThat(nyeAndeler[0].stønadTom).isEqualTo(forrigeAndelTom)
+            assertThat(nyeAndeler[0].beløp).isEqualTo(50)
+            assertThat(nyeAndeler[1].stønadFom).isEqualTo(nyAndelFom)
+            assertThat(nyeAndeler[1].stønadTom).isEqualTo(nyAndelTom)
+            assertThat(nyeAndeler[1].beløp).isEqualTo(100)
+            assertThat(nyeAndeler[0].kildeBehandlingId).isNotEqualTo(nyeAndeler[1].kildeBehandlingId)
```

Comment:
Burde man kunne sjekke att de ikke er endret med bare en sjekk att listene er like? 
```suggestion
            assertThat(nyeAndeler).containsExactlyElementsOf(forrigeAndeler + beløpsperioder)
```